test/csuite/wt8963_insert_stress/main.c

```diff
@@ -0,0 +1,171 @@
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * The motivation for this test is to try and reproduce BF-24385 by stressing insert functionality.
+ * The test creates a lot of threads that concurrently insert a lot of records with random keys.
+ * Having a large memory page ensures that we have big insert lists. Big cache size allows having
+ * more dirty content in the memory before eviction kicks in.
+ */
+
+#define THREAD_NUM_ITERATIONS 200000
+#define NUM_THREADS 110
+#define KEY_MAX UINT32_MAX
+
+void *thread_insert_race(void *);
+
+static uint64_t ready_counter;
+static WT_RAND_STATE rnd;
+
+/*
+ * set_key --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_key variadic argument.
+ */
+static void
+set_key(WT_CURSOR *c, uint64_t value)
+{
+    c->set_key(c, value);
+}
+
+/*
+ * set_value --
+ *     Wrapper providing the correct typing for the WT_CURSOR::set_value variadic argument.
+ */
+static void
+set_value(TEST_OPTS *opts, WT_CURSOR *c, uint64_t value)
+{
+    if (opts->table_type == TABLE_FIX)
+        c->set_value(c, (uint8_t)value);
+    else
+        c->set_value(c, value);
+}
+
+/*
+ * main --
+ *     Test's entry point.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    WT_CURSOR *cursor;
+    WT_SESSION *session;
+    clock_t ce, cs;
+    pthread_t id[NUM_THREADS];
+    int i, ret;
+    char tableconf[128];
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    opts->nthreads = NUM_THREADS;
+    opts->table_type = TABLE_ROW;
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+    testutil_make_work_dir(opts->home);
+
+    testutil_check(
+      wiredtiger_open(opts->home, NULL, "create,cache_size=3G,statistics=(fast)", &opts->conn));
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &session));
+    testutil_check(__wt_snprintf(tableconf, sizeof(tableconf),
+      "key_format=%s,value_format=%s,leaf_page_max=32k,memory_page_image_max=50MB",
+      opts->table_type == TABLE_ROW ? "Q" : "r", opts->table_type == TABLE_FIX ? "8t" : "Q"));
+    testutil_check(session->create(session, opts->uri, tableconf));
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    cs = clock();
+
+    /* Multithreaded insert */
+    for (i = 0; i < (int)opts->nthreads; ++i)
+        testutil_check(pthread_create(&id[i], NULL, thread_insert_race, opts));
+
+    while (--i >= 0)
+        testutil_check(pthread_join(id[i], NULL));
+
+    /* Reopen connection for WT_SESSION::verify. It requires exclusive access to the file. */
+    testutil_check(opts->conn->close(opts->conn, NULL));
+    opts->conn = NULL;
+    testutil_check(wiredtiger_open(opts->home, NULL,
+      "create,cache_size=4G,eviction=(threads_max=1),statistics=(fast)", &opts->conn));
```

Comment:
Why are we setting the eviction threads and statistics here?

Reply:
Sorry, it somehow sneaked into the PR. I removed it and somehow it's still here. Here's the fix: 6f1a58fae7623a6cbed1cb31f166025f6b4550b8

Reply:
Now I understand what's happened here. Classic code duplication issue. I had 2 connection strings. Fixed 625e4e2414502ba28920041ba2362f5c93890708

Reply:
And regarding statistics, I thought that it would be good to have it if we get a crash.

Reply:
I see that, but statistics are shared cache things, and so I think they're likely to impact (by some unknown amount), thread independence.

Reply:
Thanks! Removed stats from the config string.